torch/nn/functional.py

```diff
@@ -1785,10 +1786,10 @@ def smooth_l1_loss(input, target, size_average=None, reduce=None, reduction='ele
     See :class:`~torch.nn.SmoothL1Loss` for details.
     """
     if size_average is not None or reduce is not None:
-        reduction = _Reduction.legacy_get_enum(size_average, reduce)
-    else:
-        reduction = _Reduction.get_enum(reduction)
-    return torch._C._nn.smooth_l1_loss(input, target, reduction)
+        reduction = _Reduction.legacy_get_string(size_average, reduce)
+    return _pointwise_loss(
```

Comment:
This looks like it's going to make smooth_l1_loss much slower. It goes from 1 kernel to something like 8.

Reply:
Yea but our TH implementation currently doesn't support `requires_grad=True` of `target`. 
@colesbury What do you recommend to solve it here? Is porting smooth_l1_loss to Aten preferred over this hacky workaround?

Reply:
I misunderstood how _pointwise_loss worked. This is fine.